File: fe/user_management/test/features/list_users/data/models/user_model_test.dart
import 'dart:convert';

import 'package:flutter_test/flutter_test.dart';
import 'package:user_management/features/list_users/data/models/user_model.dart';

import '../../../../fixtures/fixture_reader.dart';

void main() {
  const UserModel tUserModel = UserModel(lastName: "Doe", firstName: "John", uuidString: "75c46e40-7098-11ed-9a71-a91a125af53c");

  group('fromJson', (){
    test('should return a valid model from JSON', (){
      final Map<String, dynamic> jsonMap =
      json.decode(fixture('user.json'));
      final result = UserModel.fromJson(jsonMap);
      expect(result, tUserModel);
    });
  });

  group('toJson', () {
    test(
      'should return a JSON map containing the proper data',
          () async {
        // act
        final result = tUserModel.toJson();
        // assert
        final Map<String, dynamic> expectedJsonMap =
        json.decode(fixture('user.json'));

        expect(result, expectedJsonMap);
      },
    );
  });
}

File: fe/kotlin/UserManagement/app/src/test/java/com/example/usermanagement/features/get_users/data/models/UserModelTest.kt
package com.example.usermanagement.features.get_users.data.models

import com.example.usermanagement.features.get_users.domain.entities.UserEntity
import com.example.usermanagement.fixtures.readFromFile
import org.hamcrest.CoreMatchers.instanceOf
import org.hamcrest.MatcherAssert.assertThat
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.junit.MockitoJUnitRunner
import java.util.*


@RunWith(MockitoJUnitRunner::class)
class UserModelTest {
    private val userModel = UserModel(uuid = UUID.fromString("75c46e40-7098-11ed-9a71-a91a125af53c"), lastName = "Doe", firstName = "John")

    @Test
    fun userModelExtendsUser() {
        assertThat(userModel, instanceOf(UserEntity::class.java))
    }

    @Test
    fun shouldReturnValidModelFromExpectedJson() {
        val expectedJson = readFromFile("user.json")
        val sut = UserModel.fromJSON(expectedJson)
        assertEquals(userModel, sut)
    }

    @Test
    fun shouldReturnExpectedStringFromToJson() {
        val expectedResult = readFromFile("user.json")
        assertEquals(expectedResult, userModel.toJson())
    }
}